### ISSUE TITLE
No-deps npm package can not download elm binary though proxy

### ISSUE THREAD

**Issue by ringvold** [elm/compiler#1963]
**Quick Summary:** Version `0.19.0-no-deps` of the npm package has problems downloading the elm binary through some http proxies.

- **Elm:** 0.19.0-no-deps
- **Browser:** N/A
- **Operating System:** Ubuntu WLS

## Additional Details

When installing elm 0.19.0-no-deps from npm we get an error:
```
-- ERROR -----------------------------------------------------------------------

I ran into trouble while fetching https://github.com/elm/compiler/releases/download/0.19.0/binary-for-linux-64-bit.gz

Error: connect ECONNREFUSED <some.ip.address.here>:443

NOTE: You can avoid npm entirely by downloading directly from:
https://github.com/elm/compiler/releases/download/0.19.0/binary-for-linux-64-bit.gz
All this package does is download that file and put it somewhere.

--------------------------------------------------------------------------------
```

We have to use a http proxy at work to do external http calls. Setting the http_proxy and http_proxy env variables in the shell and/or in npm works in most situations but not here.

I have been digging and it seems like if one use Nodes internal http and https libraries it is a bit more complicated to get a https request through a http proxy. See the links below.
Apparently one needs to create a http connection to the http proxy and then do the https request through that connection.

Most recommendations is to use some library, like `request`, which handles proxies automatically. As this is a no-deps version one would need to handle this manually to avoid this problem.

The part of the installer that would need change is [requestWithRedirects function](https://github.com/elm/compiler/blob/6173f080396a2f4c0215898ae4b68c7f22252aac/installers/npm/download.js#L73). The the current request would need to be wrapped with a `http.request` (not https) using host and port (and potentially auth; might be a bit more complicated) from the proxy environment variables.
See links below for example and some more information.

I can add a suggested fix in a PR if that is desirable. :)

https://stackoverflow.com/a/49611762
https://github.com/nodejs/node-v0.x-archive/issues/2474#issuecomment-3481078

**Comment by evancz** [comment]
Thank you for the report! I need to do some further testing of the `npm` installer before 0.19.1 comes out, so I can look into how the HTTPS_PROXY environment variable can be accounted for.

I'd like to keep this package with zero deps, and based on my initial looking around, it seems like that will be plausible to maintain.

**Comment by evancz** [comment]
This should be resolved with `npm install elm@latest-0.19.1`

Can you confirm?

**Comment by ringvold** [comment]
Yes it works now! :)
